core/datatrak.c

```diff
@@ -616,9 +616,10 @@ static void wlog_compl_parser(struct memblock *wl_mem, struct dive *dt_dive, int
 	 * Extended notes string. Fixed length 256 bytes. 0 padded if not complete
 	 */
 	if (*(runner + offset)) {
-		wlog_notes = calloc(strlong + 1, 1);
-		wlog_notes = memcpy(wlog_notes, runner + offset, 256);
-		wlog_notes = to_utf8((unsigned char *) wlog_notes);
+		char *wlog_notes_temp = calloc(strlong + 1, 1);
```

Comment:
This one is a bit weird. `strlong` appears to be a constant 256. Which means that this could be allocated on the stack. Then in the next line the 256 is hardcoded!

For now this fix is fine, but this deserves some closer look.